solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -843,6 +847,31 @@ public void downloadFromZK(String zkPath, Path dir) throws IOException {
     ZkMaintenanceUtils.downloadFromZK(this, zkPath, dir);
   }
 
+  /**
+   * Ensures we can subsequently read the most up-to-date state of the provided {@code path} and all data below
+   * it, after this method completes.
+   * This should not be called a lot; there may be some delay.  Consider alternative approaches It's often better to try to devise a way to
+   * "watch" for state changed instead of calling this.
+   */
+  public void sync(String path) {
+    // zookeeper.sync is asynchronous; we need to wait till it's done
+    CompletableFuture<Integer> future = new CompletableFuture<>();
+    keeper.sync(path, (rc, path1, ctx) -> future.complete(rc), null);
+    try {
+      Integer status = future.get(zkClientTimeout, TimeUnit.MILLISECONDS);
```

Comment:
should this method return if the sync was successful or not?

Reply:
I guess I could do that but it's only caller doesn't care.  It's a best-effort kind of thing.  I should add that to the javadocs; this method should not throw an exception.

Reply:
Dumb question from something who hasn't dug into ZK client code in Java in ages, but if you're waiting on the answer anyway, why use a future and then wait it instead of just calling a synchronous method to begin with?  Or is more like, "I'm waiting on potentially lots and lots of answers, I just need it to be up to date with any previous operations"?

Reply:
Why synchronous method do you refer to?   If there was one that accomplishes this task here, I'd call it!  keeper.sync() is an asynchronous method.